vmdk_plugin/drivers/vmdk/vmdkops/esx_vmdkcmd.go

```diff
@@ -34,7 +34,12 @@ import (
 
 /*
 #cgo CFLAGS: -I ../../../../esx_service/vmci
-#include "vmci_client.c"
+#cgo windows LDFLAGS: -L. -lvmci_client
+#ifdef _WIN32
+	#include "vmci_client.h"
+#else
+	#include "vmci_client.c"
```

Comment:
1. vmci_client.h needs to have standard double-inclusion guard (regardless of anything else)
```c
#ifndef __VMCI_CLIENT__
#define __VMCI_CLIENT__

... actual .h content ...
#endif /* __VMCI_CLIENT__ */

```

2. nothing prevents vmci_client.c to have #ifndef WIN32 for the code you don't like in windows,. 
and go would just include vmci_client.c

3. I suspect a better structure could be done still, but the above would also work